### PR TITLE
use transactions and handle unit discrepencies

### PR DESCRIPTION
**Pull request by maripuncel**
@mpuncel

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -3,6 +3,7 @@ var express = require('express'),
   bodyParser = require('body-parser'),
   mysql = require('mysql'),
   path = require('path'),
+	bookshelf = require('./database/db'),
```
Comment (mpuncel) [inline server.js]: indentations be weird yo

File: server.js [hunk 2]
```diff
@@ -30,52 +31,62 @@ app.post('/items', function (req, res) {
...
-        });
-      }
-    });
-  });
-  Promise.all(promises).then(function() {
-    return res.send('Success');
-  });
+
+	bookshelf.transaction(function(t) {
+		promises = itemList.map(function(item) {
```
Comment (mpuncel) [inline server.js]: indentations be weird yo

File: server.js [hunk 3]
```diff
@@ -59,22 +65,27 @@ app.delete('/items', function (req, res) {
...
       });
     });
-  });
-  Promise.all(promises).then(function() {
-    return res.send('Success');
+    return Promise.all(promises).then(t.commit).catch(t.rollback);
+  }).then(function(success) {
+    res.send('SUCCESS');
+  }).catch(function(failure) {
+    res.status(400).send(failure);
```
Comment (maripuncel) [inline server.js:88]: could do a better job of parsing failures here....

File: server.js [hunk 4]
```diff
@@ -30,27 +31,32 @@ app.post('/items', function (req, res) {
...
+    promises = itemList.map(function(item) {
+      return Item.where('name', item.name).fetch().then(function(results){
+        if (results === null) {
+          return Item.forge({name: item.name, quantity: item.quantity, unit: item.unit}).save(null, {transacting: t}).then(function(r) {
+            var itemId = r.get('id');
+            return PurchasedItem.forge({item_id: itemId, quantity: item.quantity, unit: item.unit, cost: 0}).save(null, {transacting: t});
           });
-        });
-      }
+        } else {
```
Comment (mpuncel) [inline server.js:46]: you could get rid of this `else` since you're returning if `results === null`

File: server.js [hunk 5]
```diff
@@ -59,22 +65,27 @@ app.delete('/items', function (req, res) {
...
-      }
-      UsedItem.forge({item_id: itemId, quantity: item.quantity, unit: 0}).save().then(function(results) {
-        if (results) return res.end('DELETE remove item');
+
+  bookshelf.transaction(function(t) {
+    promises = itemList.map(function(item) {
+      return Item.where('name', item.name).fetch().then(function(results){
+        if (results === null) {
+          return Promise.reject(`No purchased item exists for ${item.name}`);
+        } if (item.unit !== results.get('unit')) {
```
Comment (mpuncel) [inline server.js]: weird lines

**Comment by mpuncel** [comment]
loooks good!

File: server.js (hunk 4 above)
Comment (maripuncel) [reply server.js:46]: Done

File: server.js (hunk 5 above)
Comment (maripuncel) [reply server.js]: YOURE A WEIRD LINE

Comment (maripuncel) [reply server.js]: but i fixed it

**Pull request closed without merging**